mesh_segmenter/src/convex_hull.cpp

```diff
@@ -0,0 +1,83 @@
+#include <mesh_segmenter/convex_hull.h>
+
+ConvexHullGenerator::ConvexHullGenerator(){}  
+
+void ConvexHullGenerator::makeMesh(const std::string& input, pcl::PointCloud<pcl::PointXYZ>& inMesh)
+{
+  //expects a .ply file
+  pcl::PLYReader Reader;
+  Reader.read(input, inMesh); //populate inMesh
+  return;
+}
+
+
+void ConvexHullGenerator::cleanMesh(const pcl::PointCloud<pcl::PointXYZ>& outMesh, pcl::PolygonMesh& outMeshPoly)
+{
+  //find centroid coords by finding average x, y, z 
+
+  Eigen::Matrix< float, 4, 1 > mid;
+  int centroid_success = pcl::compute3DCentroid(outMesh, mid);
+  Eigen::Vector3d midVec= {mid[0], mid[1], mid[2]};
+
+  //invert bad polygons --------------------------------
+  for (int t=0; t < (outMeshPoly.polygons.size()); t++)
```

Comment:
is there a compiler warning here? why the extra brackets? Perhaps you need a std::static_cast<int>(blablabla)